### DISCUSSION TITLE
Piecewise Bilinear Interpolation of Variable Values

### DISCUSSION THREAD

**Discussion by AndrewFalkowski** [idaholab/moose#24031] [Q&A General] [answered]
Hello, 

I am wondering if there is a way to compute a piecewise bilinear output based on variables other than spatial coordinates and time? I have a boundary that varies as a function of pressure and time; however, the variation cannot be defined with an analytical expression and must be interpolated. Using a function BC that pulls a value from a `PiecewiseBilinear` function seems like the right approach, but I don't think there is a way to interpolate within `PiecewiseBilinear` with quantities other than space and time as it is currently written. Is there a way to accomplish this without writing a custom kernel?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Typically, variable dependence is not for Functions. Functions are pre-determined functions of time and space. 
You ll want to look at a material property instead. If it s just for output (and only output) you can consider an auxkernel too. 

If the dependencies are simple, you can use a ParsedMaterial. 
If they are complicated, you ll want to code but there are good utilities for performing these operations
These utilities are shared across different systems. In PiecewiseBilinear function, you can look at the _bilinear_interp member of the class, it s something you can create in your custom material to do the interpolations

Guillaume
